### PR TITLE
Add configuration for Travis CI

### PR DESCRIPTION
**Pull request by acant**
Since Travis now has support for OSX images, it might be helpful to get the CI tests running with it.
I am also going to have a look at [wwtd](https://github.com/grosser/wwtd) (what would travis do?) and see if that might be useful to running the same tests locally

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  - gem install bundler
+  - gem update bundler
+rvm:
+  - 2.4.1
+  - jruby-1.7.26
+  - jruby-9.1.8.0
+  - rbx-3.72
+os:
+  - osx
+osx_image: xcode7.3.1
```
Comment (ttilley) [inline .travis.yml]: I generally don't care about this early of a version of xcode

File: Rakefile [hunk 2]
```diff
@@ -2,6 +2,8 @@
 require 'bundler'
 Bundler::GemHelper.install_tasks
 
+require 'wwtd/tasks'
```
Comment (ttilley) [inline Rakefile]: where is this coming from?

Comment (ttilley) [inline Rakefile]: why use a separate gem for travis stuff? what's going on here that i'm missing?

File: .travis.yml (hunk 1 above)
Comment (ttilley) [inline .travis.yml]: the test runner bits also use rbenv instead of rvm. is this possible here?

**Comment by acant** [comment]
Hi @ttilley thanks for the feedback and sorry for the delay.

I reviewed the available OSX versions, and I will update the config to use 'osx_image: xcode9' instead of 7.3.1
https://docs.travis-ci.com/user/languages/objective-c#Supported-OS-X-iOS-SDK-versions

The 'rvm' configuration is a Travis thing, and it does not appear to support rbenv.
https://docs.travis-ci.com/user/languages/ruby/#Choosing-Ruby-versions-and-implementations-to-test-against

Finally, the wwtd gem is not needed for Travis itself, but in theory will execute the tests locally based upon the Travis configuration. I was trying it out but was not quite able to get the different version of ruby installed the way it liked. IF it works it could avoid duplicating the list of ruby version to against in the travis.yml and Rakefile.
I will drop it from the branch for the moment and just get Travis working. I'll keep playing with it locally and see what is necessary to make it work.

**Comment by ttilley** [comment]
reopening then. if it doesn't require rvm locally that'd be pretty sweet.

**Comment by acant** [comment]
Finally ready to merge @ttilley 
The build is green https://travis-ci.org/acant/rb-fsevent/builds/267430582
but rubinius does not yet appear to be supported in the TravisCI OSX environment. I have included in the test matrix, but allow it to fail so that it will start as soon as they do add it.
(There current list is here https://docs.travis-ci.com/user/reference/osx/#Ruby-versions-implementations)t

**Comment by thibaudgg** [comment]
Closing that one, I think now it would make more sense to use GitHub actions directly, but waiting for the Mac M1 architecture could be a good idea.

**Pull request closed without merging**